playbooks/infra/deploy-vm-bastion-libvirt.yml

```diff
@@ -0,0 +1,241 @@
+
+    - name: Customize qcow2 VM image
+      ansible.builtin.command:
+        "virt-customize -a {{ libvirtd_disk_path }}/{{ vm_disk_name }} --root-password password:{{ ansible_password }}
+         --ssh-inject root:string:\"{{ ssh_public_key }}\""
+      register: resize_output
+      changed_when: resize_output.rc == 0
+
+    - name: Create bastion VMs
+      ansible.builtin.import_role:
+        name: redhatci.ocp.create_vms
+      vars:
+        additional_virt_install_options: "--cloud-init network-config=/tmp/{{ vm_name }}-network.yaml --wait=0"
+        cluster_name: "dummy"
+        vm_bridge_name: default
+        images_dir: "{{ libvirtd_disk_path }}"
+        kvm_nodes:
+          - name: "{{ vm_name }}"
+            disks:
+              main: "1" # Value ignored because hdd created on the previous step.
+            memory: "{{ hostvars['bastion'].vm_ram }}"
+            network_interfaces:
+              "{{ {hostvars['bastion'].vm_bridge_name: ''} }}"
+            uuid: ""
+            vcpu: "{{ hostvars['bastion'].vm_cpu }}"
+
+- name: Initial system setup
+  hosts: bastion
+  gather_facts: false
+  vars:
+    ansible_user: "{{ initial_user }}"
+    system_rpm_path: "/tmp/{{ system_rpm_link | basename }}"
+  tasks:
+    - name: Wait 1200 seconds for target connection to become reachable/usable
+      ansible.builtin.wait_for_connection:
+        delay: 30
+        sleep: 10
+        timeout: 1200
+
+    - name: Get system rpm from repository
+      ansible.builtin.get_url:
+        url: "{{ system_rpm_link }}"
+        dest: "{{ system_rpm_path }}"
+        force: false
+        mode: "0640"
+
+    - name: Install system rpm
+      ansible.builtin.dnf:
+        name: "{{ system_rpm_path }}"
+        state: present
+        disable_gpg_check: true
+
+    - name: Activate OS
+      ansible.builtin.command:
+        "{{ activate_system_cmd }}"
+      changed_when: false
+
+    - name: "Deploy user {{ hostvars[inventory_hostname].ansible_user }}"
+      ansible.builtin.user:
+        name: "{{ hostvars[inventory_hostname].ansible_user }}"
+        create_home: true
+        groups: wheel
+        password: "{{ hostvars[inventory_hostname].ansible_password | password_hash('sha512') }}"
+        state: present
+
+    - name: Set paswordless sudo
+      ansible.builtin.lineinfile:
+        path: /etc/sudoers.d/{{ hostvars[inventory_hostname].ansible_user }}
+        line: "{{ hostvars[inventory_hostname].ansible_user }} ALL=(ALL) NOPASSWD: ALL"
+        mode: "0640"
+        create: true
+
+    - name: Remove cloud-init default sshd config
+      ansible.builtin.file:
+        path: "/etc/ssh/sshd_config.d/50-cloud-init.conf"
+        state: absent
+
+    - name: Restart SSH daemon
+      ansible.builtin.systemd:
+        name: sshd
+        state: restarted
+
+    - name: Run partprobe to update kernel partition table
+      ansible.builtin.command:
+        cmd: partprobe
+      changed_when: true
+
+- name: Setup bastion host
```

Comment:
Looks like we might need another user for connecting in the pipeline, currently it was passwordless sudo.
Maybe we can add the option to the playbook?

Reply:
It's OK. We can always modify it once we have use case for multiple users. WDYT?

Reply:
I think the use case is to have a different user for CI than the one for management